implementation/GUI/gui.py

```diff
@@ -21,6 +22,10 @@ def __init__(self, backend):
         # things in the graphics area
         self.image = None  # qtg.QPixmap
 
+        # two types of images
+        self.np_image = None
```

Comment:
idk if np_image and mp_image should be in GUI when they are clearly a backend responsibility.